### PR TITLE
Standalone build

### PR DESCRIPTION
**Pull request by dummdidumm**
See #239  for more

### CODE REVIEW

**Comment by dummdidumm** [comment]
This works as standalone now when using it with a bundler like Rollup. But it doesn't work yet for a true browser import because there's a `prettier` and `svelte/compiler` import inside the standalone ESM. These need to get out of there somehow, because imports like that only work with import maps currently which are not standard yet and only present in chromium based browsers.

**Comment by raymondmuller** [comment]
@dummdidumm Can this get merged (once conflicts are resolved)?
Even though it doesn't work 'yet' as a true browser import due to the dependencies, you mention it does work when bundled, which IMO is a great addition already (and I could use this now 😄  )

File: package.json [hunk 1]
```diff
@@ -3,9 +3,26 @@
...
+                "import": "./esm/standalone.mjs",
+                "default": "./standalone.js"
+            },
+            "default": "./plugin.js"
+        },
+        "./standalone": {
+            "import": "./esm/standalone.mjs",
+            "default": "./standalone.js"
+        },
+        "./package.json": "./package.json"
```
Comment (benmccann) [inline package.json:19]: I think we updated `rollup-plugin-svelte`, so there's no reason to export this anymore

File: rollup.config.js [hunk 2]
```diff
@@ -1,14 +1,63 @@
 import resolve from '@rollup/plugin-node-resolve';
 import commonjs from '@rollup/plugin-commonjs';
 import typescript from 'rollup-plugin-typescript';
+import alias from '@rollup/plugin-alias';
```
Comment (benmccann) [inline rollup.config.js:4]: this should probably go before the others to be alphabetized?

File: rollup.config.js [hunk 3]
```diff
@@ -1,14 +1,63 @@
...
 
-export default {
-    input: 'src/index.ts',
-    plugins: [resolve(), commonjs(), typescript()],
-    external: ['prettier', 'svelte'],
-    output: {
-        file: 'plugin.js',
-        format: 'cjs',
-        sourcemap: true,
+const srcDir = path.resolve(__dirname, 'src');
```
Comment (benmccann) [inline rollup.config.js:7]: do we want to use svelte naming convetions?
```suggestion
const src_dir = path.resolve(__dirname, 'src');
```

File: src/index.ts [hunk 4]
```diff
@@ -25,7 +26,8 @@ export const parsers: Record<string, Parser> = {
     svelte: {
         parse: (text) => {
             try {
-                return <ASTNode>{ ...require(`svelte/compiler`).parse(text), __isRoot: true };
+                // @ts-ignore
```
Comment (benmccann) [inline src/index.ts:29]: should add a comment why this is necessary. if unfixable should use `@ts-expect-error`

**Comment by curran** [comment]
See also https://github.com/sveltejs/prettier-plugin-svelte/pull/417

**Pull request closed without merging**